app/views/unattended/provisioning_templates/PXEGrub2/kickstart_default_pxegrub2.erb

```diff
@@ -32,11 +32,12 @@ menuentry '<%= template_name %>' {
 
 <%
 # Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1811561 and https://bugzilla.redhat.com/show_bug.cgi?id=1842893
-if @host.provision_interface.subnet.httpboot
-  proxy_http_port = @host.provision_interface.subnet.httpboot.httpboot_http_port
-  proxy_https_port = @host.provision_interface.subnet.httpboot.httpboot_https_port
+subnet = @host.provision_interface.subnet || @host.provision_interface.subnet6
+if subnet && subnet.httpboot
+  proxy_http_port = subnet.httpboot.httpboot_http_port
```

Comment:
This can raise an exception if HTTP is disabled on the Smart Proxy. Do we want to gracefully handle that and not generate `EFI HTTP` in that case? This is the default production deployment so a lot of users will hit it.

Note that this is not explained very clearly in the exception.

https://github.com/theforeman/foreman/blob/0108a314f02a2780732219996c3d4bf2b9ab8fc3/app/models/smart_proxy.rb#L103-L109

The same goes for the HTTPS port but there's less of a problem since that's enabled by default.

Reply:
We shouldn't have rescue in the template itself and we don't seem to have a method falling gracefully. The best way would be to add such method and allow it in safe mode instead of these two.

Reply:
Guys, I have a fresh Fedora installation and I can't test this end to end, so I asked Radovan to test this last proposal. This is the last beta blocker, let's push this out of doors. He is testing a16e161b709a86f20d9e5961143362bafef875a7.

To your point (which I can fix in another PR if needed), isn't the purpose of this code to actually raise an exception? I do not understand what the expectation is in this case.

Reply:
@ekohl actually don't we have enabled [both](https://github.com/theforeman/puppet-foreman_proxy/blob/476dfbeefda001ae297d24c4e1940c589c8435af/manifests/params.pp#L225) by default? If that's the case, I'd be OK with improving this in a separate PR.

Reply:
We have 9090 always turned on and 8000 only if Templates or HTTPBoot features are turned on.

For the record, this code is a workaround for grub2 bug. It will be removed sometimes in future.

Reply:
Only if HTTP is on, which is off by default: https://github.com/theforeman/puppet-foreman_proxy/blob/476dfbeefda001ae297d24c4e1940c589c8435af/manifests/params.pp#L138

https://bugzilla.redhat.com/show_bug.cgi?id=1857263#c2 has a more detailed analysis.

> To your point (which I can fix in another PR if needed), isn't the purpose of this code to actually raise an exception? I do not understand what the expectation is in this case.

In 2.2 it's not as much a problem since we disable TFTP by default and thus also HTTPBoot, but since you want to cherry pick to 2.1 this is a problem. There the current end result is that in a current **default** deployment the template doesn't render because you get an exception. Currently we turn HTTPBoot on by default but the Smart Proxy doesn't listen on HTTP.

I also opened https://github.com/theforeman/puppet-foreman_proxy/pull/608 to disable HTTPBoot even if you enable TFTP to make it explicit opt in.

For the short term I would suggest to change the methods I linked and split them in `httpboot_http_port` (which can return nil) and `httpboot_http_port!` (which raises an exception). Same for https. Then in the template you can deal with nil by not rendering the block.

The upside of this is that it works and the user can't reach an invalid state. The downside is that the user might not be aware that they're missing out on features. This can be mitigated by rendering a comment
```erb
<% if httpboot_http_port -%>
  HTTP BOOT PART HERE
<% else -%>
  # HTTPBoot over HTTP is unavailable on #{smart_proxy}
<% end -%>
```

Reply:
That's right, this has a potential to break things. I like this approach more, actually the patch change is quite small. I amended a change, there are new two helper macros and the template now uses version which returns `nil`. In that case, a comment is rendered so we don't break anything for anybody.